internal/venice-client-common/src/main/java/com/linkedin/venice/utils/VeniceProperties.java

```diff
@@ -427,26 +427,22 @@ public Map<String, String> getMap(String key) {
     if (!containsKey(key)) {
       throw new UndefinedPropertyException(key);
     }
-    Map<String, String> map = new HashMap<>();
     List<String> keyValuePairs = this.getList(key);
-
-    for (String pair: keyValuePairs) {
-      Pair<String, String> keyValuePair = splitAnEntryToKeyValuePair(pair);
-      map.put(keyValuePair.getFirst(), keyValuePair.getSecond());
-    }
-    return map;
-  }
-
-  private Pair<String, String> splitAnEntryToKeyValuePair(String entry) {
-    if (!entry.contains(":")) {
-      throw new VeniceException("Invalid config. Expect each entry to contain at least one \":\". Got: " + entry);
+    Map<String, String> map = new HashMap<>(keyValuePairs.size());
+
+    for (String keyValuePair: keyValuePairs) {
+      // One entry could have multiple ":". For example, "<ID>:<Kafka URL>:<port>". In this case, we split the String by
```

Comment:
can we use `keyValuePair.split(":", 2)` to perform this ?

Reply:
Yes, we could write it like:

```
      String[] keyValueArray = keyValuePair.split(":", 2);
      if (keyValueArray.length != 2) {
        throw new VeniceException(
            "Invalid config. Expect each entry to contain at least one \":\". Got: " + keyValuePair);
      }
      String mapKey = keyValueArray[0];
      String mapValue = keyValueArray[1];
      map.put(mapKey, mapValue);
```

Is it more clear? I am not sure...

Reply:
I was expecting using `split` will make it more compact. But does not look like it :(

Reply:
Well it could be more compact if we inline the array access like so:

```
map.put(keyValueArray[0], keyValueArray[1]);
```

Although our line length limit would also allow inlining the previous function calls as well...

```
map.put(keyValuePair.substring(0, indexOfFirstColon), keyValuePair.substring(indexOfFirstColon + 1));
```

But either way, there still needs to be a check for whether the colon is completely missing...